qiskit/transpiler/passes/routing/stochastic_swap.py

```diff
@@ -333,13 +355,130 @@ def _mapper(self, circuit_graph, coupling_graph, trials=20):
+            dag_block = circuit_to_dag(block)
+            _pass = self.__class__(new_coupling, initial_layout=None, seed=self.seed)
+            updated_dag_block = _pass.run(dag_block)
+            block_dags.append(updated_dag_block)
+            block_layouts.append(_pass.property_set["final_layout"].copy())
+            # block_layouts.append(_pass.property_set["final_layout"].copy())
+        changed_layouts = [current_layout != layout for layout in block_layouts]
+        if not any(changed_layouts):
+            return cf_op, current_layout
+        depth_cnt = [bdag.depth() for bdag in block_dags]
+        maxind = np.argmax(depth_cnt)
+        for i, dag in enumerate(block_dags):
+            if i == maxind:
+                block_circuits.append(dag_to_circuit(dag))
+            else:
+                layout_xform = LayoutTransformation(
+                    new_coupling,
+                    block_layouts[i],
+                    block_layouts[maxind],
+                )
+                match_dag = layout_xform.run(dag)
+                block_circuits.append(dag_to_circuit(match_dag))
+
+        final_permutation = combine_permutations(
+            list(current_layout.get_physical_bits().keys()),
+            list(block_layouts[maxind].get_physical_bits().keys()),
+        )
+        final_layout = Layout.from_intlist(final_permutation, *self.qregs.values())
+        return cf_op.replace_blocks(block_circuits), final_layout
+
+    def _transpile_controlflow_looping(self, cf_op, current_layout):
+        """for looping this pass adds a swap layer at the end of the loop body to bring
+        the layout back to the expected starting layout. This could be reduced a bit by
+        specializing to for_loop and while_loop. If the loop body contains a "continue_loop", the
+        swap layer is placed before the continue". No check is done to make sure "continue" is final
+        statement in loop block.
+        """
+        # pylint: disable=cyclic-import
+        from qiskit.transpiler.passes.routing import LayoutTransformation
+
+        def _move_continue_to_end(dag):
+            """Check if continue exists in block. If it does move it to the end of the circuit
+            so layout_xform doesn't get skipped."""
+            continue_nodes = dag.named_nodes("continue_loop")
+            if continue_nodes:
+                if len(continue_nodes) > 1:
+                    raise CircuitError("Multiple 'continue' statements contained in loop")
+                if len(continue_nodes) <= 1:
+                    c_node = continue_nodes[-1]
+                    dag.remove_op_node(c_node)
+                    dag.apply_operation_back(c_node.op, c_node.qargs, c_node.cargs)
+            return dag
+
+        new_coupling = layout_transform(self.coupling_map, current_layout)
+        dag_block = circuit_to_dag(cf_op.blocks[0])
+        _pass = self.__class__(new_coupling, initial_layout=None, seed=self.seed)
```

Comment:
By default the seed is `None`. I'm not sure of a reason for it not to be except for deterministic testing?

Part of the reason for the class is because it stores the `initial_layout` as an attribute. In the interest of keeping the meaning of that the same in the handling of control flow blocks I didn't change how this is referenced.

Reply:
The coupling_map no longer changes as of 53231fbc5 but the meaning of `initial_layout` and `final_layout` still do. Perhaps it is possible to avoid this but maybe we can revisit it later if it seems to cause performance issues.